source/connect/connection-options.txt

```diff
@@ -0,0 +1,356 @@
+       | **Connection URI Example**: ``readPreference=secondaryPreferred``
+   
+   * - :manual:`maxStalenessSeconds </reference/connection-string/#mongodb-urioption-urioption.maxstalenessseconds>`
+     - | **Data Type**: {+int-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = [ 'maxStalenessSeconds' => 30 ];``
+       | **Connection URI Example**: ``maxStalenessSeconds=30``
+
+   * - :manual:`readPreferenceTags </reference/connection-string/#mongodb-urioption-urioption.readpreferencetags>`
+     - | **Data Type**: ``array``
+       | **MongoDB\\Client Example**:
+       
+       .. code-block:: php
+
+          $uriOptions = [
+            'readPreferenceTags' => [
+              [ 'region' => 'east' ],
+              [ 'region' => 'west']
+            ]
+          ];``
+        
+       **Connection URI Example**: ``readPreferenceTags=region:east,region:west``
+
+Authentication Options
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Connection Option
+     - Description
+   
+   * - :manual:`authMechanism </reference/connection-string/#mongodb-urioption-urioption.authmechanism>`
+     - | **Data Type**: {+string-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = [ 'authMechanism' => 'MONGODB-X509' ];``
+       | **Connection URI Example**: ``authMechanism=MONGODB-X509``
+
+   * - :manual:`authMechanismProperties </reference/connection-string/#mongodb-urioption-urioption.authmechanismproperties>`
+     - | **Data Type**: {+string-data-type+}
+       | **MongoDB\\Client Example**:
+       | ``$uriOptions = [ 'authMechanismProperties' => 'AWS_SESSION_TOKEN:12345' ];``
+       | 
+       | **Connection URI Example**: ``authMechanismProperties=AWS_SESSION_TOKEN:12345``
+
+   * - :manual:`authSource </reference/connection-string/#mongodb-urioption-urioption.authsource>`
+     - | **Data Type**: {+string-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = [ 'authSource' => 'admin' ];``
+       | **Connection URI Example**: ``authSource=admin``
+
+Server Selection and Discovery Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 30 70
+
+   * - Connection Option
+     - Description
+   
+   * - :manual:`localThresholdMS </reference/connection-string/#mongodb-urioption-urioption.localthresholdms>`
+     - | **Data Type**: {+int-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = [ 'localThresholdMS' => 20 ];``
+       | **Connection URI Example**: ``localThresholdMS=20``
+
+   * - :manual:`serverSelectionTimeoutMS </reference/connection-string/#mongodb-urioption-urioption.serverselectiontimeoutms>`
+     - | **Data Type**: {+int-data-type+}
+       | **MongoDB\\Client Example**: ``$uriOptions = [ 'serverSelectionTimeoutMS' => 40000 ];``
+       | **Connection URI Example**: ``serverSelectionTimeoutMS=40000``
```

Comment:
```suggestion
       | **MongoDB\\Client Example**: ``$uriOptions = [ 'serverSelectionTimeoutMS' => 5000 ];``
       | **Connection URI Example**: ``serverSelectionTimeoutMS=5000``
```

PHP defaults to 30000 (i.e. 30 seconds), which is already quite high but comes from a time when replica set elections took a lot longer. I'd suggest using 5000 here. Even though we're not explaining the specifics, it'd be helpful to present something more reasonable to the user.

Most applications would want to tune this value smaller to "fail fast" anyway -- particularly PHP apps.